Fix inventory locks

trasher opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

trasher commented:
Sorry for the late last commit, I should have seen that before.

cedric-anne commented on src/Inventory/Asset/Peripheral.php:
```diff
@@ -163,10 +162,13 @@ public function handle()
                     $items_id = $data['found_inventories'][0];
+                    $peripheral->getFromDB($items_id);
+                    $handled_input = $this->handleInput($val, $peripheral);
+                    //FIXME: handleInput
```
```suggestion
```

cedric-anne commented on tests/functionnal/Lockedfield.php:
```diff
@@ -602,4 +602,309 @@ public function testLockedDdValue()
+        //check created databases & instances
+        $this->integer(countElementsInTable(\DatabaseInstance::getTable()))->isIdenticalTo(1);
+
+        //ensure database version has been updated
```
```suggestion
        //make sure manufacturer is still the correct one
```

cedric-anne reviewed (approved):
I'm a bit lost after all these changes made on locks management, but it seems OK.

trasher commented:
> I'm a bit lost after all these changes made on locks management, but it seems OK.

So am I :D We've added tests, we fixed all that together with @stonebuzz; and requester has confirmed this fix his issues.

stonebuzz commented:
Just a remark (this will be part of another PR)
a lock on the OS architecture is not displayed in the "lock" tab

![image](https://user-images.githubusercontent.com/7335054/191290989-476ceba4-05a8-4ede-bda9-a9c6704f5af4.png)

![image](https://user-images.githubusercontent.com/7335054/191291012-fdf62816-e4dc-4ac1-ae71-5bdf6a8bac6c.png)

trasher commented:
> Just a remark (this will be part of another PR) a lock on the OS architecture is not displayed in the "lock" tab

Lock tab shows only locks for current itemtype... You will only see locks on current computer, never OS devices, etc. But indeed, that is anothher problem, so another PR ;)
